### PR TITLE
ROC-6130: Inversion of Control for legal representatives frontend

### PR DESCRIPTION
**Pull request by sabahirfan**
This PR implements inversion of Control for legal representatives frontend.
### JIRA ###
[ROC-6130](https://tools.hmcts.net/jira/browse/ROC-6130)

### CODE REVIEW

**Comment by klawani** [comment]
This repos PRs are failing integration test at the moment in AKS Preview due to authentication with IDAM AAT. 
I am currently investigation to fix the issue. 
I have an opened PR: https://github.com/hmcts/cmc-legal-rep-frontend/pull/466
for the fix.

File: src/main/app/claims/claimStoreClient.ts [hunk 1]
```diff
@@ -6,21 +6,31 @@ import { ClaimModelConverter } from 'claims/claimModelConverter'
...
       body: convertedDraftClaim,
       headers: {
         Authorization: `Bearer ${user.bearerToken}`
       }
     })
   }
 
+  static getUri (user: User) {
+    if (toBoolean(config.get('featureToggles.inversionOfControl'))) {
+      return `${claimStoreApiUrl}/${user.id}/create-legal-rep-claim`
```
Comment (simad) [inline src/main/app/claims/claimStoreClient.ts]: we could've passed the user id directly, but 👍

File: infrastructure/variables.tf [hunk 2]
```diff
@@ -44,6 +44,10 @@ variable "feature_certificateOfService" {
   default = "false"
 }
 
+variable "feature_inversionOfControl" {
```
Comment (simad) [inline infrastructure/variables.tf]: this mixed camel / snake case is triggering my _eyetwitch_. I know it's not your fault as there's already `certificateOfService` breaking this convention, should we change that too?

Comment (sabahirfan) [reply infrastructure/variables.tf]: Agree, but trying to make it consistent with rest of the flags. This is temporary flag which we will remove after go live.

File: src/main/app/claims/claimStoreClient.ts (hunk 1 above)
Comment (sabahirfan) [reply src/main/app/claims/claimStoreClient.ts]: 👍

**Pull request merged**